.github/workflows/psalm.yml

```diff
@@ -0,0 +1,61 @@
+name: Static analysis
+
+on:
+  pull_request: ~
+
+jobs:
+  psalm:
+    name: Psalm
+    runs-on: Ubuntu-20.04
+
+    steps:
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.0'
+          extensions: "json,couchbase,memcached,mongodb,redis,xsl,ldap,dom"
+          ini-values: "memory_limit=-1"
+          coverage: none
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Configure composer
+        run: |
+            COMPOSER_HOME="$(composer config home)"
+            composer self-update
+            ([ -d "$COMPOSER_HOME" ] || mkdir "$COMPOSER_HOME") && cp .github/composer-config.json "$COMPOSER_HOME/config.json"
+            echo "COMPOSER_ROOT_VERSION=$(grep -m1 SYMFONY_VERSION .travis.yml | grep -o '[0-9.x]*').x-dev" >> $GITHUB_ENV
+
+      - name: Determine composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: composer-${{ github.sha }}
+          restore-keys: composer-
+
+      - name: Cache Psalm
+        uses: actions/cache@v2
+        with:
+          path: .github/.cache/psalm/
+          key: psalm-${{ github.sha }}
+          restore-keys: psalm-
+
+      - name: Install dependencies
+        run: |
+            echo "::group::modify composer.json"
+            sed -i -re 's/"replace": \{/"replace": \{"symfony\/phpunit-bridge": "self.version",/' composer.json
+            composer require --no-update psalm/phar:4.6.1 phpunit/phpunit phpspec/prophecy php-http/discovery
```

Comment:
Do we really need Prophecy? I wasn't aware that we actually use it for anything.

Reply:
It is just the one place: https://github.com/symfony/symfony/blob/5.2/src/Symfony/Component/ErrorHandler/DebugClassLoader.php#L307

Maybe I should remove this.